drivers/dai/intel/uaol/uaol_regs_ace1x.h

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __INTEL_DAI_DRIVER_UAOL_REGS_ACE1X_H__
+#define __INTEL_DAI_DRIVER_UAOL_REGS_ACE1X_H__
+
+#include <stdint.h>
+
+/* UAOL Shim registers */
+#define UAOLCAP_OFFSET				0x00
+#define UAOLCTL_OFFSET				0x04
+#define UAOLIPPTR_OFFSET			0x08
+#define UAOLxPCMSCAP_OFFSET			0x10
+#define UAOLxPCMSyCM_OFFSET(y)			(0x12 + 0x2 * (y))
+#define UAOLxPCMSyCHC_OFFSET(y)			(0x1E + 0x2 * (y))
+
+/* UAOL IPx Registers */
+#define UAOLxTBDF_OFFSET			0x00
+#define UAOLxSUV_OFFSET				0x02
+#define UAOLxOPC_OFFSET				0x04
+#define UAOLxIPC_OFFSET				0x06
+#define UAOLxFC_OFFSET				0x10
+#define UAOLxFA_OFFSET				0x14
+#define UAOLxIC_OFFSET				0x18
+#define UAOLxIR_OFFSET				0x1C
+#define UAOLxICPy_OFFSET(y)			(0x20 + 0x04 * (y))
+#define UAOLxIRPy_OFFSET(y)			(0x30 + 0x04 * (y))
+#define UAOLxPCMSyCTL_OFFSET(y)			(0x40 + 0x20 * (y))
+#define UAOLxPCMSySTS_OFFSET(y)			(0x48 + 0x20 * (y))
+#define UAOLxPCMSyRA_OFFSET(y)			(0x4C + 0x20 * (y))
+#define UAOLxPCMSyFSA_OFFSET(y)			(0x50 + 0x20 * (y))
+
+/* ALH registers */
+#define STRMzTXDA_OFFSET(z)			(0x000 + 4 * (z))
+#define STRMzRXDA_OFFSET(z)			(0x100 + 4 * (z))
+
+union UAOLCTL {
```

Comment:
This seems to be a bit excessive. There seems to be only difference in this UAOLCTL struct and UAOLxPCMSyCM_OFFSET -- all in all ~10 definitions, and rest of the 200 lines of defs are the same.